client/src/components/profile.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import {
+  Container, ListGroup, Row, Col,
+} from 'react-bootstrap';
+import {
+  Link, Switch, Route, Redirect,
+} from 'react-router-dom';
+import PropTypes from 'prop-types';
+import Info from './profile/info';
+import History from './profile/history';
+import Orders from './profile/orders';
+import Settings from './profile/settings';
+import Sharing from './profile/sharing';
+import Subscriptions from './profile/subscriptions';
+import Api from '../services/api';
+import '../style/profile.css';
+
+class Profile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {},
+      isFetching: false,
+    };
+    this.menuItems = [
+      { path: '/profile/info', title: 'Info' },
+      { path: '/profile/orders', title: 'Orders' },
+      { path: '/profile/history', title: 'History' },
+      { path: '/profile/subscriptions', title: 'Subscriptions' },
+      { path: '/profile/sharing', title: 'Sharing' },
+      { path: '/profile/settings', title: 'Settings' },
+    ];
+  }
+
+  async componentDidMount() {
+    this.getProfile();
+  }
+
+  getProfile() {
+    Api.getOne('persons', 2)
+      .then((response) => {
+        if (response.error) {
+          // eslint-disable-next-line no-console
+          console.error(response);
+          return;
+        }
+        this.setState({
+          user: response.data,
+          isFetching: true,
+        });
+      });
+  }
+
+  render() {
+    const { isFetching, user } = this.state;
+    const { location } = this.props;
+    return (
+      <Container fluid>
+        <Container fluid className="page-header">
+          <h1 className="page-header-title">
+            Profile
+          </h1>
+        </Container>
+        <Container className="card-body pl-5 pr-5">
+          <Row>
+            <Col md="3" className="profile-menu">
+              <ListGroup>
+                {this.menuItems.map((menuItem, index) => {
+                  return (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Link to={menuItem.path} key={index}>
```

Comment:
What 'index' refers to here?

Reply:
In this case index is index of array element https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
I use it because of that `menuItems` array is static.

https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318
> Update: Exception from the rule
> 
> Many people asked if they always, always have to generate ids. Others have suggested use cases when using the index as a key seems justifiable.
> 
> It is true that sometimes generating new ids is redundant and may be avoided. For example translation of license terms or list of contributors.
> 
> To help you decide, I put together three conditions which these examples have in common:
> 
>     the list and items are static–they are not computed and do not change;
>     the items in the list have no ids;
>     the list is never reordered or filtered.
> 
> When all of them are met, you may safely use the index as a key.

Reply:
Nice. Thank you!